app/lti.go

```diff
@@ -212,8 +211,16 @@ func (a *App) generateFreeUsername(baseUsername string) (string, *model.AppError
 	}
 
 	// Base username is not free, append an incrementing number until we find a free username
-	// Truncate the baseUsername to the max username length - suffix for appended numbers
-	baseUsername = baseUsername[:model.UserNameMaxLength-len(strconv.Itoa(MaxUsernameSuffix))]
+	// Make sure that the baseUsername with an appended suffix does not exceed the max length
+	// of a username
+	// Note a slice greater than the length of the thing sliced is an error, so the baseUsername only
+	// gets truncated if needed.
+	var maxBaseUsernameLen = model.UserNameMaxLength - len(strconv.Itoa(MaxUsernameSuffix))
+	var basenameLen = int64(math.Min(float64(len(baseUsername)), float64(maxBaseUsernameLen)))
```

Comment:
OK. I'm not so sure about the conversion to floats to use min and then converting back to an int, but it did reduce the lines of code and is clear.